packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -36,284 +37,116 @@ const
-			if (prevProps.spotlightId !== this.props.spotlightId) {
-				this.configureSpotlight(this.props.spotlightId);
-			}
-			this.restoreFocus();
-		}
 
-		componentWillUnmount () {
-			const containerNode = this.uiRefCurrent.containerRef.current;
-			const scrollerNode = document.querySelector(`[data-spotlight-id="${this.props.spotlightId}"]`);
+			// componentWillUnmount
+			return () => {
+				if (type === JS) {
+					// remove a function for preventing native scrolling by Spotlight
+					if (containerNode && containerNode.removeEventListener) {
+						containerNode.removeEventListener('scroll', variables.current.preventScroll);
+					}
+				}
 
-			if (type === JS) {
-				// remove a function for preventing native scrolling by Spotlight
-				if (containerNode && containerNode.removeEventListener) {
-					containerNode.removeEventListener('scroll', this.preventScroll);
+				if (scrollerNode && scrollerNode.removeEventListener) {
+					scrollerNode.removeEventListener('keydown', onKeyDown, {capture: true});
+					scrollerNode.removeEventListener('keyup', onKeyUp, {capture: true});
 				}
-			}
 
-			if (scrollerNode && scrollerNode.removeEventListener) {
-				scrollerNode.removeEventListener('keydown', this.onKeyDown, {capture: true});
-				scrollerNode.removeEventListener('keyup', this.onKeyUp, {capture: true});
-			}
+				pause.resume();
+				SpotlightAccelerator.reset();
 
-			this.pause.resume();
-			SpotlightAccelerator.reset();
+				setContainerDisabled(false);
+			};
+		}, []);	// TODO : Handle exhaustive-deps ESLint rule.
 
-			this.setContainerDisabled(false);
-		}
+		// componentDidUpdate
+		useEffect(() => {
+			configureSpotlight(props.spotlightId);
+		}, [props.spotlightId]);	// TODO : Handle exhaustive-deps ESLint rule.
 
-		isScrolledBy5way = false
-		isScrolledByJump = false
-		isWrappedBy5way = false
-		lastFocusedIndex = null
-		nodeIndexToBeFocused = null
-		preservedIndex = null
-		restoreLastFocused = false
-		uiRefCurrent = null
+		useEffect(restoreFocus);	// TODO : Handle exhaustive-deps ESLint rule.
 
-		setContainerDisabled = (bool) => {
-			const
-				{spotlightId} = this.props,
-				containerNode = document.querySelector(`[data-spotlight-id="${spotlightId}"]`);
+		function setContainerDisabled (bool) {
+			const containerNode = document.querySelector(`[data-spotlight-id="${spotlightId}"]`);
 
 			if (containerNode) {
 				containerNode.setAttribute(dataContainerDisabledAttribute, bool);
 
 				if (bool) {
-					document.addEventListener('keydown', this.handleGlobalKeyDown, {capture: true});
+					document.addEventListener('keydown', handleGlobalKeyDown, {capture: true});
 				} else {
-					document.removeEventListener('keydown', this.handleGlobalKeyDown, {capture: true});
+					document.removeEventListener('keydown', handleGlobalKeyDown, {capture: true});
 				}
 			}
 		}
 
-		configureSpotlight = (spotlightId) => {
-			const {spacing} = this.props;
+		function configureSpotlight () {
```

Comment:
Why did you omit the `spotlightId` parameter here?

Reply:
`spotlightId` is declared in outer scope also .(line 65). 
So, we don't need to receive the parameter here.